samples/Beyer2024/create_paper_data.py

```diff
@@ -0,0 +1,187 @@
+# Import pyMBE and other libraries
+import pyMBE
+from lib import analysis
+import os 
+import numpy as np
+import pandas as pd
+import argparse 
+
+# Create an instance of pyMBE library
+pmb = pyMBE.pymbe_library()
+
+valid_fig_labels=["6a", "6b", "6c"]
+valid_modes=["short-run","long-run", "test"]
+
+parser = argparse.ArgumentParser(description='Script to create the data from Beyer2024')
+parser.add_argument('--fig_label', 
+                    type=str, 
+                    required= True,  
+                    help=f'Label of the corresponding figure in Beyer2024, currently supported: {valid_fig_labels}')
+parser.add_argument('--mode', 
+                    type=str, 
+                    default= "long-run",  
+                    help='Sets for how long the simulation runs, valid modes are {valid_modes}')
+parser.add_argument('--restart', action='store_true', help="Switch to plot the data")
+args = parser.parse_args()
+
+# Inputs
+fig_label=args.fig_label
+mode=args.mode
+plot=args.plot
+print(f"plot {args.plot}")
+# Sanity checks
+if fig_label not in valid_fig_labels:
+    raise ValueError(f"The figure label {fig_label} is not supported. Supported figure labels are {valid_fig_labels}")
+
+
+if mode not in valid_modes:
+    raise ValueError(f"Mode {mode} is not currently supported, valid modes are {valid_modes}")
+
+## Peptide plots (Fig. 6)
+labels_fig6=["6a", "6b", "6c"]
+
+if fig_label in labels_fig6:
+    script_path=pmb.get_resource(f"samples/Beyer2024/peptide.py")
+    if fig_label == "6a":
+        sequence="K"*5+"D"*5
+    elif fig_label == "6b":
+        sequence="E"*5+"H"*5
+    elif fig_label == "6c":
+        sequence="nDSHAKRHHGYKRKFHEKHHSHRGYc"
+    else:
+        raise RuntimeError()
+    pH_range = np.linspace(2, 12, num=21)
+    for pH in pH_range:
+        run_command=f"python3 {script_path} --sequence {sequence} --pH {pH} --mode {mode}"
+        print(run_command)
+        os.system(run_command)
```

Comment:
Here and elsewhere (including in files not affected by this PR): I would **strongly** recommend moving away from [`os.system()`](https://docs.python.org/3/library/os.html#os.system) (which directly exposes the shell), in favor of [`subprocess.check_output()`](https://docs.python.org/3/library/subprocess.html#subprocess.check_output) without `shell=True`. Rules for escaping characters with special meaning in paths (e.g. whitespace, slashes) are cumbersome and easy to forget; `subprocess` does all that for you. When a shell command fails, `os.system` doesn't care: it returns an OS-specific error code and Python executes the next line, possibly one of these `os.system(f"rm {path}/*")` which might not delete the files you expected if `path` depended on a failed shell command...

MWE:

```py
import os
import subprocess
os.system("ls important_file.bak") # is there a backup file?
print("rm important_file")         # oops!
subprocess.check_output(["ls", "important_file.bak"])
print("rm important_file")         # will never execute without a backup file
```
Output:
```
ls: cannot access 'important_file.bak': No such file or directory
rm important_file
ls: cannot access 'important_file.bak': No such file or directory
Traceback (most recent call last):
  File "/home/user/mwe.py", line 5, in <module>
    subprocess.check_output(["ls", "important_file.bak"])
  File "/usr/lib/python3.10/subprocess.py", line 421, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['ls', 'important_file.bak']' returned non-zero exit status 2.
```

Note that `subprocess.run()` would not throw an exception, since it returns an object to wrap the exception and provide more context, like the error code and the command line arguments, to help the user diagnose the issue. Also, some of the `rm path/*` commands can probably be replaced with a temporary folder context manager, which is automatically cleaned up when leaving the context.

Reply:
@jngrad if you do not mind that we deal with the sub-process refactoring in a separated PR, then I would proceed with this PR so we can have the new CI in main for continuing with the rest of the work.  I know that the tree structure is still not the best, I only wanted to have a clean structure for the article before we designed what you suggested in #13. If you are OK then we can merge this PR once you approve it.